### ISSUE TITLE
afterburn seems to keep appending to authorized_keys.d/afterburn

### ISSUE THREAD

**Issue by dustymabe** [coreos/fedora-coreos-tracker#441]
I'm testing out a dev build with `afterburn-4.3.2-1.fc31.x86_64` on gce. I notice that the ssh authorized keys seems to have many duplicate entries which indicates to me that afterburn is just appending to the file over and over:

```
[bound] -bash-5.0$ cat /home/core/.ssh/authorized_keys.d/afterburn | wc -l
17
[bound] -bash-5.0$ 
[bound] -bash-5.0$ cat /home/core/.ssh/authorized_keys.d/afterburn | sort | uniq | wc -l 
3
```

Is this expected behavior?

I started this instance with kola spawn:

```
kola spawn -p gce --gce-project fedora-coreos-devel --gce-json-key /srv/key.json --gce-image fedora-coreos-fedora-coreos-31-20200330-dev-1-gcp-x86-64
```

**Comment by lucab** [comment]
@dustymabe interesting. No, that's not the expected behavior. Afterburn should be [stomping on](https://github.com/coreos/afterburn/blob/v4.3.2/src/providers/mod.rs#L140) on the existing file.

I have a fear that the result you is just mirroring the fact that you indeed have 17 SSH keys in your project. Can you double-check what's the status of your project with regards to keys in your GCP console?

**Comment by dustymabe** [comment]
hey @lucab - I do believe you are right that this was a problem with the environment. This was a brand new project and the GCP web interface was not properly reporting back to me that SSH keys had been saved when I added them to the project, so I kept adding them (still the web interface didn't tell me any ssh keys were in the project). Since I could never add an ssh key to the project (in the web UI) I used kola. Apparently all the times I had tried to add a key to the project it did work in the web interface, it just wasn't showing me that they had been added. I thought it was just me being dumb because I'm new to GCP but I shared my screen with benjamin and neither one of us could figure out what was going on. Today I go into the web interface and a bunch of keys show up in the project wide keys metadata, so I guess it was an intermittent issue.
